src/ranch.erl

```diff
@@ -85,6 +87,48 @@ normalize_opts(Map) when is_map(Map) ->
 normalize_opts(Any) ->
 	#{socket_opts => Any}.
 
+-spec validate_transopts(transport_opts(any())) -> ok | {error, any()}.
+validate_transopts(Opts) ->
+	maps:fold(fun
```

Comment:
Maybe define most of the fun as a real function instead of indenting like mad.

```
validate_transport_opt(connection_type, worker) -> true;
validate_transport_opt(connection_type, supervisor) -> true;
%% ...
validate_transport_opt(_, _) -> false.
```